integ_test/contract.js

```diff
@@ -31,46 +34,76 @@ function deployCt(ct, ctData, args, cb) {
         //.on('confirmation', (confirmationNumber, receipt) => { console.log(confirmationNumber, receipt) });
 }
 
-
 describe('contracts', () => {
   let opts = { 
       from: test_cfg.TEST_ACCT_ADDR,
       gas: test_cfg.GAS,
       gasPrice: test_cfg.GAS_PRICE,
   };
 
-  let typesBin
-  let typesAbi
   let ct;
   let ctInstAddress;
 
+  let web3 = new Web3(client);
+
+  let typesSrc;
+  let compiled;
+  let BIN;
+  let ABI;
+  let contract;
+
   let testNumber = 31337;
+  let source = fs.readFileSync(typesSrcPath, 'utf-8');
 
   before(done => {
     if( test_cfg.TEST_ACCT_ADDR.length == 0 ) { 
         done(Error("Error during setup.  No test account address was configured."));
     }
 
-    let steps = {
-      typesBin: async.apply(fs.readFile, typesBinPath),
-      typesAbi: async.apply(fs.readFile, typesAbiPath),
+    let steps = {    
+      typesSrc: async.apply(fs.readFile, typesSrcPath),
+
+      compiled: ['typesSrc', async.apply(function (res, cb){
+          let compiled2 = web3.eth.compileSolidity(source, function(err,res){ //necessary callback function
+            cb(null, res);
+          });
+
+      })],
+
+      BIN: ['compiled', async.apply(function(res, cb){
```

Comment:
let's rename BIN to bin to be consistent with the other names in this Object

Reply:
good idea